compiler/rustc_mir_dataflow/src/impls/single_enum_variant.rs

```diff
@@ -0,0 +1,215 @@
+use super::*;
+//use crate::lattice::PackedU8JoinSemiLattice as Fact;
+
+//use crate::lattice::FactArray;
+use crate::lattice::FactCache;
+use rustc_target::abi::VariantIdx;
+
+use rustc_middle::mir::*;
+
+use crate::{Analysis, AnalysisDomain};
+
+/// A dataflow analysis that tracks whether an enum can hold exactly 1, or more than 1 variants.
+///
+/// Specifically, if a variable is constructed with a value of `Some(1)`,
+/// We should be able to optimize it under the assumption that it has the `Some` variant.
+/// If a variable can be multiple variants, then we assume nothing.
+/// If a variable
```

Comment:
(looks like you left out the end of a sentence). When you say variable here, are you talking specifically about locals? What is the exact conclusion I can draw from this analysis telling me that the variant of a local `i` is `V` at some point in the control flow. I believe it should be something like "the result of a `Discriminant(i)` computation, if it is not UB, will be the discriminant of `V`"

Reply:
Oops I think I just copied the sentence above I have no idea what I meant.
At a given point in time, if local `i` is associate with `V`, that implies `Discriminant(i) = V`